deep_agent/aegra/thread_cleanup.py

```
@@ -0,0 +1,159 @@
+
+
+async def _delete_checkpoints(thread_id: str) -> int:
+    """Delete all LangGraph checkpoint data for a thread."""
+    try:
+        import psycopg
+
+        async with await psycopg.AsyncConnection.connect(settings.database_uri) as conn:
+            deleted = 0
+            for table in ("checkpoint_writes", "checkpoint_blobs", "checkpoints"):
+                cur = await conn.execute(
+                    f"DELETE FROM {table} WHERE thread_id = %s",
```

Ruff (0.16.0)

[error] Possible SQL injection vector through string-based query construction

(S608)